File: src/main/java/at/hrechny/predictionsbot/database/entity/CompetitionEntity.java
package at.hrechny.predictionsbot.database.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import java.util.List;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@EqualsAndHashCode(callSuper = true)
@Entity
@Table(name = "competitions")
public class CompetitionEntity extends GeneratedIdEntity {

  @Column
  private String name;

  @Column(unique=true)
  private Long apiFootballId;

  @OneToMany(mappedBy="competition")
  private List<SeasonEntity> seasons;

}

File: src/main/kotlin/at/hrechny/predictionsbot/database/entity/CompetitionEntity.kt
package at.hrechny.predictionsbot.database.entity

import io.micronaut.core.annotation.Introspected
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.OneToMany
import jakarta.persistence.Table

@Introspected(accessKind = [Introspected.AccessKind.FIELD])
@Entity
@Table(name = "competitions")
class CompetitionEntity : GeneratedIdEntity() {
    @field:Column
    var name: String? = null

    @field:Column(unique = true)
    var apiFootballId: Long? = null

    @field:OneToMany(mappedBy = "competition")
    var seasons: MutableList<SeasonEntity> = mutableListOf()
}
